src/lerobot/utils/queue.py

```diff
@@ -29,11 +30,22 @@ def get_last_item_from_queue(queue: Queue, block=True, timeout: float = 0.1) ->
     else:
         item = None
 
-    # Drain queue and keep only the most recent parameters
-    try:
-        while True:
-            item = queue.get_nowait()
-    except Empty:
-        pass
+    if platform.system() == "Darwin":
+        # On Mac, avoid using `qsize` due to unreliable implementation.
+        # There is a comment on `qsize` code in the Python source:
+        # Raises NotImplementedError on Mac OSX because of broken sem_getvalue()
+        try:
+            while True:
+                item = queue.get_nowait()
+        except Empty:
+            pass
+    else:
+        # Drain queue and keep only the most recent parameters
+        while queue.qsize():
+            try:
+                while True:
+                    item = queue.get_nowait()
+            except Empty:
+                pass
```

Comment:
```suggestion
    # Drain queue and keep only the most recent parameters
    if platform.system() == "Darwin":
        # On Mac, avoid using `qsize` due to unreliable implementation.
        # There is a comment on `qsize` code in the Python source:
        # Raises NotImplementedError on Mac OSX because of broken sem_getvalue()
        try:
            while True:
                item = queue.get_nowait()
        except Empty:
            pass
            
        return item
        
        while queue.qsize():
            with suppress(Empty): item = queue.get_nowait()
```

Wdyt about such approach? The code looks cleaner, it will also required to import `from contextlib import suppress
` in to of the file.

Reply:
Also, to be sure that it works - please run tests locally, and probably we will need one/two tests with `from torch.multiprocessing import Event, Queue` queue in tests, for now it uses just Queue.

Tests are located there `tests/utils/test_queue.py`